comm/src/main/scala/coop/rchain/comm/transport/GrpcTransportReceiver.scala

```diff
@@ -39,7 +44,8 @@ object GrpcTransportReceiver {
       buffersMap: Ref[Task, Map[PeerNode, Deferred[Task, MessageBuffers]]],
       messageHandlers: MessageHandlers,
       tempFolder: Path,
-      parallelism: Int
+      parallelism: Int,
+      chunks: TrieMap[String, Array[Byte]]
```

Comment:
We might need to put a constraint on block size or this Map can potentially exceed Heap size

Reply:
Agree, we also need to investigate how to get more info about the message before it's completely received. Streams can have some header announcing the length.